README.md

```diff
@@ -20,135 +14,123 @@ protocol ProtocolName {
 
 ```js
 protocol ToString {
   tag;
 
   toString() {
-    return `[object ${this[ToString.tag]}]`;
+    return `[object ${this[ToString].tag}]`;
   }
-}
-
-Object.prototype[ToString.tag] = 'Object';
-Protocol.implement(Object, ToString);
-```
-
-The auto-flattening behaviour of `Promise.prototype.then` was a very controversial decision.
-Valid arguments exist for both the auto-flattening and the monadic versions to be the default.
-Protocols eliminate this issue in two ways:
-
-1. Symbols are unique and unambiguous. There is no fear of naming collisions,
-   and it is clear what function you are using.
-1. Protocols may be applied to existing classes, so there is nothing
-   preventing consumers with different goals from using their own methods.
 
-```js
-protocol Functor {
-  map;
-}
-
-class Identity {
-  constructor(val) { this.val = val; }
-  unwrap() { return this.val; }
+  // Coherence-guaranteed implementation for existing classes
+  implFor Object {
+    tag = 'Object';
+  };
 }
-
-Promise.prototype[Functor.map] = function (f) {
-  return this.then(function(x) {
-    if (x instanceof Identity) {
-      x = x.unwrap();
-    }
-    let result = f.call(this, x);
-    if (result instanceof Promise) {
-      result = new Identity(result);
-    }
-    return result;
-  });
-};
-
-Protocol.implement(Promise, Functor);
 ```
 
-Finally, one of the biggest benefits of protocols is that they eliminate the
-fear of mutating built-in prototypes. One of the beautiful aspects of
-ECMAScript is its ability to extend its built-in prototypes. But with the
-limited string namespace, this is untenable in large codebases and impossible
-when integrating with third parties. Because protocols are based on symbols,
-this is no longer an anti-pattern.
+Protocols isolate implementations meant for different purposes by keeping them
+in a separate object with certain static guarantees.
+
+One of the biggest benefits of protocols is that they eliminate the
+fear of mutating built-in prototypes in potentially conflicting ways. One of the
+beautiful aspects of ECMAScript is its ability to extend its built-in
+prototypes. But with the limited string namespace, this is untenable in large
+codebases and impossible when integrating with third parties. Because protocols
+are based on symbols and support coherence, this is no longer an anti-pattern.
 
 ```js
 class Ordering {
-  static LT = new Ordering;
-  static EQ = new Ordering;
-  static GT = new Ordering;
+  constructor (str) { this.symbol = Symbol(str) }
 }
 
 protocol Ordered {
+  static LT = new Ordering('LT');
+  static EQ = new Ordering('EQ');
+  static GT = new Ordering('GT');
+  static fromNum (num) {
+      if (num < 0) return Ordered.LT;
+      if (num === 0) return Ordered.EQ;
+      if (num > 0) return Ordered.GT;
+    }
+  }
   compare;
 
   lessThan(other) {
-    return this[Ordered.compare](other) === Ordering.LT;
+    return this[Ordered].compare(other) === Ordered.LT;
+  }
+
+  implFor String {
```

Comment:
This seems to be encouraging mutating existing prototypes quite a bit more than the current proposal; is that intentional?

Reply:
The existing proposal does not have a very different stance on this. My point with this stuff is that implementing protocols on existing prototypes is safe.